### PR TITLE
Upgrade to React 15 and improve build slightly

### PR DESCRIPTION
**Pull request by thejameskyle**

### CODE REVIEW

File: example/base.css [hunk 1]
```diff
@@ -30,7 +30,7 @@ body {
 }
 
 .cf-example-content {
-  padding: 0 2rem;
+  padding: 0 2rem 3rem;
```
Comment (thejameskyle) [inline example/base.css]: This was driving me crazy

File: lerna.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "lerna": "2.0.0-beta.13",
+  "lerna": "2.0.0-beta.20",
```
Comment (thejameskyle) [inline lerna.json]: Had to upgrade lerna to support all the different ways of specifying versions.

`npm run bootstrap` gets much faster as a result

File: gulpfile.babel.js [hunk 3]
```diff
@@ -147,6 +148,14 @@ function initBrowserify(files, output, externals, requires, transforms, watch) {
     bundler.require(requires);
   }
 
+  if (!watch) {
+    bundler.transform(envify({
+      NODE_ENV: 'production'
```
Comment (thejameskyle) [inline gulpfile.babel.js]: Better minifying

File: packages/cf-builder-card/package.json [hunk 4]
```diff
@@ -12,16 +12,16 @@
     "cf-component-card": "^0.2.1",
     "cf-component-dynamic-content": "^1.0.4",
     "cf-util-markdown": "^2.0.0",
-    "react": "^0.14.2",
+    "react": "^0.14.2 || ^15.0.0-0",
```
Comment (thejameskyle) [inline packages/cf-builder-card/package.json]: Allowing both versions to keep this from being breaking changes

File: packages/cf-builder-pagination/package.json [hunk 5]
```diff
@@ -11,12 +11,12 @@
...
-    "react": "^0.14.2"
+    "react": "^0.14.2 || ^15.0.0-0"
   },
   "devDependencies": {
     "cf-test-stub": "^1.0.1",
     "enzyme": "^2.3.0",
-    "react-addons-test-utils": "^0.14.2",
-    "react-dom": "^0.14.2"
+    "react-addons-test-utils": "^15.0.0-0",
+    "react-dom": "^15.0.0-0"
```
Comment (thejameskyle) [inline packages/cf-builder-pagination/package.json]: `devDeps` can all be just 15 though

File: gulpfile.babel.js [hunk 6]
```diff
@@ -13,6 +13,7 @@ const buffer = require('vinyl-buffer');
 const eslint = require('gulp-eslint');
 const uglify = require('gulp-uglify');
 const mkdirp = require('mkdirp');
+const envify = require('envify/custom');
```
Comment (terinjokes) [inline gulpfile.babel.js]: why not `loose-envify`?

File: packages/cf-component-modal/example/basic/component.js [hunk 7]
```diff
@@ -55,11 +55,26 @@ class Component extends React.Component {
...
+    return (
+      <GatewayProvider>
+        <div>
+          {this.props.children}
+          <GatewayDest name="modal"/>
+        </div>
+      </GatewayProvider>
+    );
+  }
+}
```
Comment (thejameskyle) [inline packages/cf-component-modal/example/basic/component.js]: Upgrading to the latest version of react-gateway is a breaking change for cf-component-modal

Comment (thejameskyle) [inline packages/cf-component-modal/example/basic/component.js]: Oh shit wait– we need to use the non-universal form of react-gateway.

Hmm– I think I'll make a new version that supports the old way of rendering in additional to universal rendering.

File: packages/cf-builder-card/package.json (hunk 4 above)
Comment (terinjokes) [inline packages/cf-builder-card/package.json]: I think everyone will thank you for the precedent of n-1 support. 😄

File: packages/cf-component-modal/example/basic/component.js (hunk 7 above)
Comment (terinjokes) [inline packages/cf-component-modal/example/basic/component.js]: > This pull request is blocked: Reason

File: gulpfile.babel.js (hunk 6 above)
Comment (thejameskyle) [inline gulpfile.babel.js]: Do you think we need to optimize this build? It only happens for the production example site. I don't expect people do anything that js-tokens can't handle. But the AST portion makes me feel safe

Comment (terinjokes) [inline gulpfile.babel.js]: I don't expect people to do anything that `js-token` wouldn't handle, but we don't currently have anything enforcing that. There's probably already a ESLint rule for it though.

**Comment by tajo** [comment]
Is this still blocked? What needs to be fixed?

**Comment by terinjokes** [comment]
@tajo I think @thejameskyle changed how `react-gateway` worked and `cf-ui` needs to be updated to support universal rendering.

**Comment by thejameskyle** [comment]
Look at the 1.x and 2.x apis of react-gateway and find a way to make them work together.

**Comment by tajo** [comment]
Can't we just major release `cf-component-modal` that will use react-gateway 2.x and forget 1.x? Or remove `cf-component-modal` from this PR. I'm not sure how to fix it but I would like to use some of cf-ui components (other than modal).

I can't even build (`npm run bootstrap`) this `react-15` branch:

```shell
~/cf-repos/cf-ui (react-15)$ npm run bootstrap

> cf-ui@ bootstrap /Users/vojtech/cf-repos/cf-ui
> lerna bootstrap

Lerna v2.0.0-beta.20
Independent Versioning Mode
Linking all dependencies
Errored while running BootstrapCommand.execute
1
```

**Comment by nkcmr** [comment]
rebased.

File: package.json [hunk 8]
```diff
@@ -67,15 +68,16 @@
     "karma-sauce-launcher": "^0.3.0",
     "karma-tape-reporter": "^1.0.3",
     "language-babel": "github:gandm/language-babel",
-    "lerna": "2.0.0-beta.13",
+    "lerna": "^2.0.0-beta.20",
```
Comment (terinjokes) [inline package.json]: @nkcmr Is it possible to upgrade to a non-beta version of lerna?

Comment (nkcmr) [inline package.json]: @terinjokes no non-beta. but there have been ~10 releases since this PR opened.

**Comment by wyuenho** [comment]
Playing with this now

**Comment by thejameskyle** [comment]
It finally happened!

**Comment by nkcmr** [comment]
noice!!

**Comment by jwineman** [comment]
God damn it, now I _HAVE_ to update partners haha.

**Pull request merged**